ast/compile.go

```diff
@@ -1353,6 +1354,32 @@ func (c *Compiler) getExports() *util.HashMap {
 	return rules
 }
 
+func (c *Compiler) checkUnusedImports() {
+	if !c.strict {
+		return
+	}
+
+	for _, name := range c.sorted {
+		mod := c.Modules[name]
+
+		for _, imp := range mod.Imports {
+			var impCounter int
+			name := imp.Name()
+
+			WalkExprs(mod, func(expr *Expr) bool {
+				if name == expr.Operand(0).Value || name == expr.Operand(1).Value {
```

Comment:
This line freaks out other tests, because expressions come in three flavours:
1. Calls -- like `eq(x, y)` -- you've got these covered here to some extent
2. Terms standalone -- like `x` which would fail if it's `undefined` or `false`
3. Every

I think there're two ways to approach this: either account for the fact that there're other types by doing something like
```go
switch e := expr.Terms.(type) {
    case []*Term:
        if name == e.Operand(0).Value || name == e.Operand(1).Value {
        // ...
        }
    case *Term: // TODO
    case *Every: // TODO
}
```

Alternatively, and I think that could be preferable, what would you think about using `WalkTerms` instead? We would not have to worry about the different types of expressions, and we'd also cover calls with more than two operands without any extra efforts.

Let's also add a few more cases:
```rego
package p
import data.foo
r = count(foo) > 1 # only one operand
```
```rego
package p
import data.foo
r = sprintf("%v %d", [foo, 0]) # more complicated operands
```
```rego
package p
import data.foo
r {
    foo # not a call, plain term
}
```
```rego
package p
import future.keywords.every
import data.foo
r {
    every x in foo { x > 1 } # import in 'every' domain
}
```
```rego
package p
import future.keywords.every
import data.foo
r {
    every x in [1,2,3] { x > foo } # import in 'every' body
}
```
I'm not completely sure what to do about this one:
```rego
package p
import data.foo
r {
    print(foo)
}
```
if print calls are enabled (in the compiler compiling that module), the import would be used; if print calls are disabled, they wouldn't.

--------

I realise that I'm asking for a bit much, so please LMK if you're up to pushing this forward; or if you'd prefer to wrap this, and have me add the last small edits. The minimum we need to do here is get the tests to pass.

Reply:
I see where you're going. Let me start making those changes then. I should have something within the next day or two.